client/src/pages/Dashboard/Dashboard.tsx

```diff
@@ -1,27 +1,48 @@
 import Box from '@material-ui/core/Box';
 import CssBaseline from '@material-ui/core/CssBaseline';
 import CircularProgress from '@material-ui/core/CircularProgress';
-import useStyles from './dashboardStyles';
 import { useAuth } from '../../context/useAuthContext';
 import { useSocket } from '../../context/useSocketContext';
 import { useHistory } from 'react-router-dom';
-import NavBar from '../../components/NavBar/NavBar';
+import { useState } from 'react';
 import { useEffect } from 'react';
+import NavBar from '../../components/NavBar/NavBar';
+import getUserBoards from '../../helpers/APICalls/getUserBoards';
+import { IBoard } from '../../interface/Board';
+import OptionsDrawer from '../../components/OptionsDrawer/OptionsDrawer';
 import Board from '../../components/Kanban/Board';
 import { KanbanProvider } from '../../context/useKanbanContext';
+import useStyles from './dashboardStyles';
 
 export default function Dashboard(): JSX.Element {
   const classes = useStyles();
-
   const { loggedInUser } = useAuth();
   const { initSocket } = useSocket();
-
+  const [openDrawer, setOpenDrawer] = useState<boolean>(false);
+  const [boards, setBoards] = useState<Array<IBoard>>([]);
   const history = useHistory();
 
+  const getAllUserBoards = async () => {
+    try {
+      const data = await getUserBoards();
+      setBoards(data.boards);
+    } catch (err) {
+      setBoards([]);
+    }
+  };
+
+  useEffect(() => {
+    getAllUserBoards();
+  }, [getAllUserBoards]);
```

Comment:
Including this in the dependencies means that every time the boards are fetched the function is rebuilt and so it is detected as a change and the API request fires forever. You could fix this in one of two ways, either using useCallback or having an empty dependency array and declaring the function inside of the useEffect

Reply:
Solved that. Thank you!